### ISSUE TITLE
When used together with `--squash`, `--iidfile` returns id of image not squashed

### ISSUE THREAD

**Issue by uzzzerrrname** [moby/moby#33822]
**Description**

When used together with `--squash`, `--iidfile` returns id of image not squashed.
This is against my expectations. See details below.

**Steps to reproduce the issue:**
1. Enable experimental features on Docker Engine
2. Build a container image with `--tag`, `--iidfile` and `--squash` options

**Describe the results you received:**
There are two images, one that is squashed and one that is not (used as input for squashing and preserved for build caching, I guess).
- `iidfile` points to the image that is *not* squashed
- `tag` is applied to the image that is squashed

**Describe the results you expected:**
`iidfile` points to the image that is squashed and `tag` is applied to the same image

**Additional information you deem important (e.g. issue happens only occasionally):**
Always happens

**Output of `docker version`:**

```
Client:
 Version:      17.06.0-ce-rc5
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   b7e4173
 Built:        Tue Jun 20 07:13:29 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.0-ce-rc5
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   b7e4173
 Built:        Tue Jun 20 07:12:36 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 10
 Running: 2
 Paused: 0
 Stopped: 8
Images: 140
Server Version: 17.06.0-ce-rc5
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.21.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 5.798GiB
Name: pprime.sm.sw.ru
ID: 4XB2:B4OK:QQED:QQO3:DNIW:LZAM:5G6U:U4KC:AZEX:RH45:Y52J:G2OE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by justincormack** [comment]
Cc @ijc

**Comment by ijc** [comment]
Thanks, Looking at it now.

**Comment by ijc** [comment]
Proposed a fix in #33824
